judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java

```diff
@@ -390,7 +390,7 @@ protected Payload insertPayloadAndAttach(EReference reference, ID identifier, Pa
                     throw new IllegalArgumentException("Upper cardinality violated");
                 }
             } else {
-                if (container.get(reference.getName()) != null) {
+                if ((container.get(reference.getName()) != null) ||  (!reference.isContainment() && !getNavigationResultAt(identifier, reference).isEmpty())) {
```

Comment:
Why does the reference being not containment (and the relation not empty) mean containment is already set?

Reply:
OK, so after some clarification apparently reference, the mapping reference is association and its mapped reference is the containment. Comment resolved